server.go

```diff
@@ -185,6 +199,25 @@ func (p *peerState) forAllPeers(closure func(p *peer)) {
 	p.forAllOutboundPeers(closure)
 }
 
+// handleUpdatePeerHeight updates the heights of all peers who were known to
+// announce a block we recently accepted.
+func (s *server) handleUpdatePeerHeights(state *peerState, umsg updatePeerHeightsMsg) {
+	state.forAllPeers(func(p *peer) {
+		// The origin peer should already have the updated height.
+		if p == umsg.originPeer {
+			return
+		}
+
+		// If the peer has recently announced a block, and this block
+		// matches our newly accepted block, then update their block
+		// height.
+		if p.lastAnnouncedBlock != nil && bytes.Equal(
+			p.lastAnnouncedBlock.Bytes(), umsg.newSha.Bytes()) {
+			p.UpdateLastBlockHeight(umsg.newHeight)
```

Comment:
Great suggestion, will skip the additional check if the peer hasn't announced a block since the last round. `lastAnnouncedBlock` now gets nilled after a peer's height has been updated. 

Nice catch, the read to `lastAnnouncedBlock` is now protected by the peer's mutex.